cmd/nerdctl/image/image_squash.go

```diff
@@ -0,0 +1,102 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package image
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/containerd/nerdctl/v2/cmd/nerdctl/helpers"
+	"github.com/containerd/nerdctl/v2/pkg/api/types"
+	"github.com/containerd/nerdctl/v2/pkg/clientutil"
+	"github.com/containerd/nerdctl/v2/pkg/cmd/image"
+)
+
+func addSquashFlags(cmd *cobra.Command) {
+	cmd.Flags().IntP("layer-count", "c", 0, "The number of layers that can be compressed")
+	cmd.Flags().StringP("layer-digest", "d", "", "The digest of the layer to be compressed")
+	cmd.Flags().StringP("author", "a", "", `Author (e.g., "nerdctl contributor <nerdctl-dev@example.com>")`)
+	cmd.Flags().StringP("message", "m", "", "Commit message")
+}
+
+func NewSquashCommand() *cobra.Command {
```

Comment:
Needs docs and tests

Reply:
ok，I will add it later.

Reply:
added, please check it out

Reply:
https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md needs to be updated

Reply:
updated, please check it out